### PR TITLE
Contract monitor configuration parameters

### PR DESCRIPTION
**Pull request by chrismaree**
This pull request enables a config object to be passed to the monitor bot. This object specifies an arrays containing addresses of liquidator and dispute bots. If an address included in these arrays creates a position, creates a liquidation or is liquidated the log text will be updated accordingly. This is not new functionality; rather this PR enables the functionality by correctly passing in the configs.

This PR also refines parameter validation in the monitor bot and adds additional comments.

### CODE REVIEW

File: monitors/index.js [hunk 1]
```diff
@@ -180,19 +176,45 @@ const Poll = async function(callback) {
     if (!process.env.EMP_ADDRESS) {
       throw "Bad environment variables! Specify an `EMP_ADDRESS` for the location of the expiring Multi Party.";
     }
-
-    if (!process.env.BOT_MONITOR_OBJECT || !process.env.WALLET_MONITOR_OBJECT) {
-      throw "Bad input arg! Specify a `BOT_MONITOR_OBJECT` & `WALLET_MONITOR_OBJECT` to track.";
+    const pollingDelay = process.env.POLLING_DELAY ? process.env.POLLING_DELAY : 300000;
+
+    if (
```
Comment (chrismaree) [inline monitors/index.js:181]: joined all these checks into one rather than having them split like before.

File: monitors/index.js [hunk 2]
```diff
@@ -180,19 +176,45 @@ const Poll = async function(callback) {
...
+      !process.env.CONTRACT_MONITOR_OBJECT ||
+      !process.env.SYNTHETIC_PEG_MONITOR_OBJECT
+    ) {
+      throw "Bad input arg! Specify a: `BOT_MONITOR_OBJECT`, `WALLET_MONITOR_OBJECT`, `CONTRACT_MONITOR_OBJECT` & `SYNTHETIC_PEG_OBJECT` to track.";
     }
 
-    const pollingDelay = process.env.POLLING_DELAY ? process.env.POLLING_DELAY : 10000;
-
-    // Bot objects to monitor. For each bot specify a name, address and the thresholds to monitor.
+    // Bots to monitor. Each bot can be have a collateralThreshold, syntheticThreshold and etherThreshold. EG:
```
Comment (chrismaree) [inline monitors/index.js:190]: added sample comments to show what the configs should look like.

File: monitors/ContractMonitor.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ class ContractMonitor {
 
   // Calculate the collateralization Ratio from the collateral, token amount and token price
   // This is cr = [collateral / (tokensOutstanding * price)] * 100
-  calculatePositionCRPercent = (collateral, tokensOutstanding, tokenPrice, price) => {
```
Comment (nicholaspai) [inline monitors/ContractMonitor.js:36]: Good catch, any idea why `price` was here before?

File: monitors/ContractMonitor.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
 const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
 
 class ContractMonitor {
-  constructor(logger, expiringMultiPartyEventClient, monitoredLiquidators, monitoredDisputers, priceFeed) {
+  constructor(logger, expiringMultiPartyEventClient, contractMonitorConfigObject, priceFeed) {
```
Comment (nicholaspai) [inline monitors/ContractMonitor.js:4]: This is cleaner +1

File: monitors/ContractMonitor.js (hunk 3 above)
Comment (chrismaree) [reply monitors/ContractMonitor.js:36]: left over oversight I guess 🤷

**Pull request merged**